test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -620,6 +621,19 @@ public void startup(ICluster cluster)
                 // Start up virtual table support
                 CassandraDaemon.getInstanceForTesting().setupVirtualKeyspaces();
 
+                // clean up debris in the rest of the keyspaces
+                for (String keyspaceName : Schema.instance.getKeyspaces())
```

Comment:
this needs to be here, actually, it should be here already. if you see what this method does, it basically copies what is in CassandraDaemon setup minus some details. I followed that method closely and this is what was missing. Without this, it will not trigger the code which scrubs data dirs / removes ephemerals.